variable/test/variable_math_test.cpp

```diff
@@ -0,0 +1,239 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+#include <scipp/variable/math.h>
+
+#include <tuple>
+
+#include <gtest/gtest.h>
+
+#include "test_macros.h"
+
+#include "scipp/core/element/math.h"
+#include "scipp/variable/variable.h"
+
+using namespace scipp;
+using namespace scipp::core;
+using namespace scipp::variable;
+
+template <typename T> class VariableMathTest : public ::testing::Test {};
+using FloatTypes = ::testing::Types<double, float>;
+TYPED_TEST_SUITE(VariableMathTest, FloatTypes);
+
+TYPED_TEST(VariableMathTest, abs) {
+  for (TypeParam x : {0.0, -1.23, 3.45, -1.23456789}) {
+    for (auto u : {units::dimensionless, units::m}) {
+      const auto v = makeVariable<TypeParam>(Values{x}, u);
+      const auto ref = element::abs(x);
+      EXPECT_EQ(abs(v), makeVariable<TypeParam>(Values{ref}, u));
```

Comment:
There is a useful shortcut for creating scalar variables: `x * u` (I believe you need to include `scipp/variable/arithmetic.h`).